src/modules/Exchange.ts

```diff
@@ -109,6 +116,87 @@ export class Exchange {
     );
   }
 
+  public async proxyDepositWithZeroEx(
```

Comment:
There will also be functions to support the ETH deposit, approval, approval with deposit?